internal/errgroup/group_test.go

```diff
@@ -0,0 +1,628 @@
+
+			g := &group{
+				enableLimitation: enableLimitation,
+				egctx:            context.Background(),
+				limitation:       make(chan struct{}),
+				errs: []error{
+					nil,
+				},
+			}
+
+			return test{
+				name: "returns nil when errs contains no error",
+				receiver: receiver{
+					g: g,
+				},
+				checkFunc: func() error {
+					if ok := g.enableLimitation.Load().(bool); ok {
+						return errors.Errorf("enableLimitation is not equals. want: %v, got: %v", false, ok)
+					}
+					return nil
+				},
+				want: nil,
+			}
+		}(),
+
+		func() test {
+			var enableLimitation atomic.Value
+			enableLimitation.Store(true)
+
+			err1 := errors.New("fail_1")
+			err2 := errors.New("fail_2")
+
+			g := &group{
+				enableLimitation: enableLimitation,
+				egctx:            context.Background(),
+				limitation:       make(chan struct{}),
+				errs: []error{
+					err1,
+					err2,
+					nil,
+				},
+			}
+
+			return test{
+				name: "returns error when errs contains error",
+				receiver: receiver{
+					g: g,
+				},
+				checkFunc: func() error {
+					if ok := g.enableLimitation.Load().(bool); ok {
+						return errors.Errorf("enableLimitation is not equals. want: %v, got: %v", false, ok)
+					}
+					return nil
+				},
+				want: errors.Wrap(
+					errors.Wrap(
+						g.err,
+						err1.Error(),
+					),
+					err2.Error(),
+				),
+			}
+		}(),
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.receiver.g.Wait()
+
+			if !errors.Is(got, tt.want) {
+				t.Errorf("not equals. want: %v, got: %v", tt.want, got)
+			}
+
+			if err := tt.checkFunc(); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestWait(t *testing.T) {
+	type global struct {
+		instance Group
+	}
+
+	type test struct {
+		name      string
+		global    global
+		checkFunc func() error
+		want      error
+	}
+
+	tests := []test{
+		func() test {
+			var enableLimitation atomic.Value
+			enableLimitation.Store(true)
+
+			g := &group{
+				enableLimitation: enableLimitation,
+				egctx:            context.Background(),
+				limitation:       make(chan struct{}),
+			}
+
+			return test{
+				name: "returns nil when instance.Wait returns nil",
+				global: global{
+					instance: g,
+				},
+				checkFunc: func() error {
+					if ok := g.enableLimitation.Load().(bool); ok {
+						return errors.Errorf("enableLimitation is not equals. want: %v, got: %v", false, ok)
+					}
+					return nil
+				},
+				want: nil,
+			}
+		}(),
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer func() {
+				clearGlobalObject()
+			}()
+
+			instance = tt.global.instance
+
+			got := Wait()
+			if !errors.Is(tt.want, got) {
+				t.Errorf("not equals. want: %v, got: %v", tt.want, got)
+			}
+
+			if err := tt.checkFunc(); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestDoCancel(t *testing.T) {
+	type receiver struct {
+		g *group
+	}
+
+	type test struct {
+		name      string
+		receiver  receiver
```

Comment:
- Could you use `g *group` instead of `receiver receiver` ?

Reply:
Thank you for your review.
Same as below.
https://github.com/vdaas/vald/pull/176/files/7e48a590487c9782db0f829d8cce1de52b3c6ab1#r376186295